qiskit_machine_learning/algorithms/classifiers/vqc.py

```diff
@@ -155,12 +165,12 @@ def fit(self, X: np.ndarray, y: np.ndarray):  # pylint: disable=invalid-name
 
         Args:
             X: The input data.
-            y: The target values.
+            y: The target values. Required to be one-hot encoded.
```

Comment:
Shall we add some input validation?

Reply:
I was thinking to open a new PR for input validation, but am happy to include it in this one!

Reply:
Given this is a bug fix, and will be backported, we try to keep the code to what is needed to correct the critical bug for users in order to minimize code change and potentially destabilize code in the stable branch via other function such as new features. I think I might count validating the input - presumably checking all the labels as a new feature/improvement. If somehow you consider it vital its with this - otherwise I would consider it preferable on a separate PR that is not backported and would be part of next main release.

Reply:
I agree to introduce new input checks in a separate PR.

Reply:
Ok. I'll open a separate issue and PR for input validation.